tests/apps/flask_app.py

```diff
@@ -212,4 +212,16 @@ def echo_use_kwargs_missing(username, **kwargs):
 def handle_error(err):
     if err.code == 422:
         assert isinstance(err.data["schema"], ma.Schema)
-    return J(err.data["messages"]), err.code
+    try:
+        return J(err.data["messages"]), err.code
+    # on marshmallow2, validation errors for nested schemas can fail to encode:
+    # https://github.com/marshmallow-code/marshmallow/issues/493
+    # to workaround this, convert integer keys to strings
+    except TypeError:
```

Comment:
I'd use `MARSHMALLOW_VERSION_INFO` here to

- ensure we are actually using MA2, not hiding another problem
- think of removing the code when dropping MA2

We can keep the try/except if you think it is better this way.

Reply:
I'll change it right now. I think being able to flag it easily when dropping MA2 support is reason enough.

Also, you're right that catching a `TypeError` could hide other things (in theory, at least).